Automattic/wp-calypso
# Purchases: add new payment method components

#### Changes proposed in this Pull Request

This PR adds new generic payment method components in preparation for us adding additional payment method options for existing purchases. A couple notes about the PR:
- Adds a new config flag `purchases/new-payment-methods`, enabled on dev, which allows us to ship regularly to production behind the scenes. 
- Updated instances of "edit" to "change" to match our UI closer. 
- Created new Add and Change payment method components that are now placed in the `manage-purchase` folder because that's where these components are accessed from. These will replace the existing Add and Edit Credit card components eventually.
- Created new `PaymentMethodForm` component which is a mirror of our current `CreditCard` form component. This is where we will begin adding more payment methods. 

**Before**
![image](https://user-images.githubusercontent.com/6981253/100790394-b6a92d80-33e5-11eb-8689-8a87c4a8f8dd.png)

**After**
![image](https://user-images.githubusercontent.com/6981253/100790796-58307f00-33e6-11eb-9187-50ef1c852e60.png)


#### Testing instructions



* Add `flags=purchases/new-payment-methods` query string if you're not testing locally
* Visit purchases screens at the site and account level
* Start by visiting the payment method tab and click "Add payment method" button. Confirm that you see the credit card form and the page title says "Add Payment Method". 
* Sandbox to confirm that you can actually add the card.
* Visit the manage purchase screen for an existing purchase that doesn't have a payment method.
* Click on "Add payment method" and confirm that you see the credit card form and the page title says "Add Payment Method".
* Sandbox to confirm that you can actually add the card.
* Visit the manage purchase screen for an existing purchase that does have a payment method attached. 
* Click on "Change payment method" and confirm that you see the credit card form and the page title says "Change Payment Method". 
* Sandbox to confirm that you can actually change the card.

Fixes https://github.com/Automattic/wp-calypso/issues/47929
